.github/workflows/publish.yml

```
@@ -1,29 +1,86 @@
-name: brew pr-pull
+name: Publish Bottles
 on:
-  pull_request_target:
-    types:
-      - labeled
+  pull_request:
+    paths:
+      - Formula/*.rb
+
 jobs:
-  pr-pull:
-    if: contains(github.event.pull_request.labels.*.name, 'pr-pull')
-    runs-on: ubuntu-22.04
+  build-bottles:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Set up Homebrew
         uses: Homebrew/actions/setup-homebrew@master
 
+      - run: brew test-bot --only-cleanup-before
+
+      - run: brew test-bot --only-setup
+
+      - run: brew test-bot --only-tap-syntax
+
+      - run: brew test-bot --only-formulae
+
+      - name: Upload bottles as artifact
+        uses: actions/upload-artifact@main
+        with:
+          name: bottles
+          path: '*.bottle.*'
+
+  brew-bottle:
+    runs-on: ubuntu-latest
+    needs: build-bottles
+    steps:
+      - name: Set up Homebrew
+        run: |
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+
+          echo "$(brew --prefix)/bin" >> $GITHUB_PATH
+
+          HOMEBREW_TAP_REPOSITORY="$(brew --repo "$GITHUB_REPOSITORY")"
+          if [[ -d "$HOMEBREW_TAP_REPOSITORY" ]]; then
+              rm -rf "$HOMEBREW_TAP_REPOSITORY"
+          fi
+          mkdir -vp "$HOMEBREW_TAP_REPOSITORY"
+
+          rm -rf "$GITHUB_WORKSPACE"
+          ln -vs "$HOMEBREW_TAP_REPOSITORY" "$GITHUB_WORKSPACE"
+
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+
       - name: Set up git
         uses: Homebrew/actions/git-user-config@master
 
-      - name: Pull bottles
-        env:
-          HOMEBREW_GITHUB_API_TOKEN: ${{ github.token }}
-          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{ github.token }}
-          HOMEBREW_GITHUB_PACKAGES_USER: ${{ github.actor }}
-          PULL_REQUEST: ${{ github.event.pull_request.number }}
-        run: brew pr-pull --debug --tap=$GITHUB_REPOSITORY $PULL_REQUEST
+      - name: Download bottle artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: bottles
+
+      - name: Merge bottles
+        run: brew bottle --merge --write --debug *.bottle.json
 
       - name: Push commits
-        uses: Homebrew/actions/git-try-push@master
+        run: |
+          git push
```

```suggestion
        run: git push
```
...unless more lines are expected to be added later.